pulp_2to3_migration/app/pre_migration.py

```diff
@@ -557,27 +579,55 @@ def mark_removed_resources(plan, type_to_repo_ids):
                                           fields=['not_in_plan'],
                                           batch_size=1000)
 
+        # Mark distributors
+        mongo_dist_object_ids = set(str(i.id) for i in Distributor.objects.only('id'))
+        premigrated_dists = Pulp2Distributor.objects.filter(
+            pulp2_repository__pulp2_repo_type=plugin_plan.type)
+        premigrated_dist_object_ids = set(premigrated_dists.values_list('pulp2_object_id',
+                                                                        flat=True))
+        removed_dist_object_ids = premigrated_dist_object_ids - mongo_dist_object_ids
+
+        removed_dists = []
+        for pulp2dist in Pulp2Distributor.objects.filter(
+                pulp2_object_id__in=removed_dist_object_ids):
+            pulp2dist.not_in_pulp2 = True
+            removed_dists.append(pulp2dist)
+
+        Pulp2Distributor.objects.bulk_update(objs=removed_dists,
+                                             fields=['not_in_plan'],
+                                             batch_size=1000)
+
 
 def delete_old_resources(plan):
     """
-    Delete pre-migrated and migrated data for resources which are migrated fully on every run.
+    Delete old Publications/Distributions which are no longer present in Pulp2.
```

Comment:
should we do same for importers?

Reply:
Not essential and there will be other orphaned resources, like old repo versions or repos themselves.